internal/topology/variables/schema.go

```diff
@@ -114,6 +114,21 @@ func convertToAPIExtensionsJSONSchemaProps(schema *clusterv1.JSONSchemaProps, fl
 		props.Minimum = &f
 	}
 
+	if schema.AdditionalProperties != nil {
+		apiExtensionsSchema, err := convertToAPIExtensionsJSONSchemaProps(schema.AdditionalProperties, fldPath.Child("additionalProperties"))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalProperties"), "",
+				fmt.Sprintf("failed to convert schema: %v", err)))
+		} else {
+			props.AdditionalProperties = &apiextensions.JSONSchemaPropsOrBool{
+				// Allows must be true to allow "additional properties".
+				// Otherwise only the ones from .Properties are allowed.
+				Allows: true,
```

Comment:
I think we don't have to support the Allow field in our types to support maps.

Upstream validation code: https://github.com/kubernetes/kube-openapi/blob/d2a55e8c85b13025de1e0bf4522860c7f20b52c7/pkg/validation/validate/object_validator.go#L68-L110

Basically if `Allows` is set to false all fields have to be defined as part of Properties (we don't support PatternProperties). So I think it's safe to assume that in our map case Allows is always true as we want to define the map value schema via `AdditionalProperties` and `AdditionalProperties` and `Properties` are mutual exclusive.

Reply:
+1 for consistency

Reply:
+1